### PR TITLE
#2058 Diff for humans milliseconds

### LINKED ISSUE #2058: Add option to include milliseconds in CarbonInterval::forHumans
Hello,

I encountered the following "issue" with the following code:
```php
$seconds = 1.114;
$diffString = Carbon::createFromFormat('U.v', round($seconds, 3))->diffForHumans(
            Carbon::createFromTimestamp(0), Carbon::DIFF_ABSOLUTE, true, 2);

echo $diffString;
```

Carbon version: 2.30.0

PHP version: 7.3.15

I want to get:

```
1s 114ms
```

But I actually get:

```
1s
```

Thanks!

I have worked out a naive solution however this is not backwards compatible and I think it'd be a useful thing to be able to specify (maybe in the options) the smallest unit allowed:
```diff
diff --git a/src/Carbon/CarbonInterval.php b/src/Carbon/CarbonInterval.php
--- a/src/Carbon/CarbonInterval.php
+++ b/src/Carbon/CarbonInterval.php
@@ -1156,10 +1156,11 @@
      $diffIntervalArray = [
            ['value' => $intervalValues->years,            'unit' => 'year',   'unitShort' => 'y'],
            ['value' => $intervalValues->months,           'unit' => 'month',  'unitShort' => 'm'],
            ['value' => $intervalValues->weeks,            'unit' => 'week',   'unitShort' => 'w'],
            ['value' => $intervalValues->daysExcludeWeeks, 'unit' => 'day',    'unitShort' => 'd'],
            ['value' => $intervalValues->hours,            'unit' => 'hour',   'unitShort' => 'h'],
            ['value' => $intervalValues->minutes,          'unit' => 'minute', 'unitShort' => 'min'],
            ['value' => $intervalValues->seconds,          'unit' => 'second', 'unitShort' => 's'],
+            ['value' => $intervalValues->milliseconds,     'unit' => 'millisecond', 'unitShort' => 'ms'],
        ];
```

### PR DESCRIPTION
**Pull request by pdscopes**
As per the discussion in #2058 this PR adds a new option `minimumUnit` to `CarbonInterval::forHumans` to allow the user to specify a minimum unit of time that they wish to be seen in the returned string. This PR simply throws away any units of time that are smaller than `minimumUnit`. This PR also includes the new option for milliseconds in the `forHumans` returned value however it defaults to seconds as the minimumUnit.

For example:

```php
CarbonInterval::setLocale('en');

echo CarbonInterval::fromString('1 second 114 milliseconds')
      ->forHumans(['parts' => 3]);
// outputs: "1 second"

echo CarbonInterval::fromString('1 hour 1 second 114 milliseconds')
      ->forHumans(['parts' => 3, 'minimumUnit' => 'hour']);
// outputs: "1 hour"

echo CarbonInterval::fromString('1 second 114 milliseconds')
      ->forHumans(['parts' => 3, 'short' => true, 'minimumUnit' => 'ms']));
// outputs: "1s 114ms"
```

This PR does not include translations into all available languages provided by Carbon and I hope that other maintainers/contributors will adds these in the future. I have attempted to translate into German, Spanish, French, and Italian but these should be checked by speakers of these languages.

Fix #2058

### CODE REVIEW

**Comment by pdscopes** [comment]
There is one more improvement I want to do to this PR which is `count($interval) === 0` to use `minimumUnit` rather than `$unit = $short ? 's' : 'second';`.

File: src/Carbon/Lang/it.php [hunk 1]
```diff
@@ -43,6 +43,9 @@
     'second' => ':count secondo|:count secondi',
     'a_second' => 'alcuni secondi|:count secondi',
     's' => ':count sec.',
+    'millisecond' => '{1}:count millisecondo|{0}:count millisecondi|]1,Inf[:count millisecondi',
+    'a_millisecond' => '{1}pochi millisecondi|{0}:count millisecondi|]1,Inf[:count millisecondi',
```
Comment (kylekatarnls) [inline src/Carbon/Lang/it.php:47]: You can rely on the `|` for the plural without specifying `{0}` `{1}`, the correct pluralization rules are provided by symfony/translation.

File: src/Carbon/CarbonInterval.php [hunk 2]
```diff
@@ -1600,6 +1606,12 @@ public function forHumans($syntax = null, $short = false, $parts = -1, $options
             if (count($interval) >= $parts) {
                 break;
             }
+
+            // break the loop after we have reached the minimum unit
+            if (in_array($minimumUnit, [$diffIntervalData['unit'], $diffIntervalData['unitShort']])) {
+                $fallbackUnit = [$diffIntervalData['unit'], $diffIntervalData['unitShort']];
+                break;
```
Comment (kylekatarnls) [inline src/Carbon/CarbonInterval.php:1613]: A blank line before statements is required by php-cs code style:
https://travis-ci.org/github/briannesbitt/Carbon/jobs/672454794#L444

File: src/Carbon/Lang/en.php [hunk 3]
```diff
@@ -36,6 +36,9 @@
     'second' => '{1}:count second|{0}:count seconds|]1,Inf[:count seconds',
     'a_second' => '{1}a few seconds|{0}:count seconds|]1,Inf[:count seconds',
     's' => ':counts',
+    'millisecond' => '{1}:count millisecond|{0}:count milliseconds|]1,Inf[:count milliseconds',
+    'a_millisecond' => '{1}a few milliseconds|{0}:count milliseconds|]1,Inf[:count milliseconds',
```
[the remark this answers is not in the document]
Comment (pdscopes) [reply src/Carbon/Lang/en.php:40]: I agree will change this to "a millisecond".

**Comment by kylekatarnls** [comment]
Thank you very much. This will be released in 2.33.

**Pull request merged**